InvenTree/report/api.py

```diff
@@ -418,13 +420,39 @@ class SalesOrderReportPrint(SalesOrderReportMixin, ReportPrintMixin, RetrieveAPI
     pass
 
 
+class ReportMetadata(RetrieveUpdateAPI):
+    """API endpoint for viewing / updating Report metadata."""
+    MODEL_REF = 'reportmodel'
+
+    def _get_model(self, *args, **kwargs):
+        """Return model depending on which report type is requested in get_view constructor."""
+        reportmodel = self.kwargs.get(self.MODEL_REF, PurchaseOrderReport)
+
+        if reportmodel not in [PurchaseOrderReport, SalesOrderReport, BuildReport, BillOfMaterialsReport, TestReport]:
+            raise ValidationError("Invalid report model")
+        return reportmodel
+
+        # Return corresponding Serializer
+    def get_serializer(self, *args, **kwargs):
+        """Return correct MetadataSerializer instance depending on which model is requested"""
+        # Get type of report, make sure its one of the allowed values
+        UseModel = self._get_model(*args, **kwargs)
```

Comment:
Can't we use a simpler approach here:

```python
UseModel = self.Meta.model
```

(or similar)?

The endpoint already has either a model or queryset defined, righ?

Reply:
This question is for @matmair, he helped out and contributed this part of the code. I'm not sure of the implications / the details of your question..

Reply:
@SchrodingersGat not in this case, the model is provided in the url definition as an argument. Reduced the boilerplate as everything is the same for these 4 models. We can type it out if you prefer that. This mainly cuts down on the LoC.